nixos/modules/services/monitoring/datadog-agent.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.datadog-agent;
+
+  ddConf = {
+    dd_url              = "https://app.datadoghq.com";
+    skip_ssl_validation = "no";
+    api_key             = "";
+    collector_log_file  = "/var/log/datadog/collector.log";
+    forwarder_log_file  = "/var/log/datadog/forwarder.log";
+    dogstatsd_log_file  = "/var/log/datadog/dogstatsd.log";
+    pup_log_file        = "/var/log/datadog/pup.log";
```

Comment:
One more thought: Most service logs on NixOS go into `journald` due to the use of systemd. Having services write logs onto disk sort of "unexpectedly" can be a bit surprising.

I don't have a strong opinion on this but it may be sensible to configure these logs to go into `journald` instead.